src/names/applications.cpp

```diff
@@ -183,4 +185,29 @@ IsValidJSONOrEmptyString (const std::string& text){
     UniValue v;
 
     return text.empty() || v.read(text);
+}    
+
+bool
+IsMinimalJSONOrEmptyString (const std::string& text){
+    UniValue v;
+    if(text.empty()){
+        return true;
+    } else if(!v.read(text)){ 
+        return false;
+    } else {
+        std::string minimalJSON = GetMinimalJSON(text);
+        LogDebug(BCLog::NAMES, "Minimalised JSON string is: %s \n", minimalJSON);
```

Comment:
I suspect that this log entry would be better if it were only done when the return value (see next line) is false, otherwise it may be a bit spammy. I'm not certain though; @domob1812 what do you think?

Reply:
Yes indeed, I think this should only be logged (if at all) for when it fails the check.

Reply:
This line should probably be indented now that it's within an `if` block?

Reply:
Possible refactor idea: create a `bool` variable called `matches`, set it to `text == minimalJSON`, then you can avoid duplicating the comparison in the `if` and the `return`. I'm guessing that compiler optimizations will produce the same ASM either way, but I think it would be a bit more readable this way.